plugins/com.google.cloud.tools.eclipse.appengine.libraries.test/src/com/google/cloud/tools/eclipse/appengine/libraries/model/LibraryFactoryTest.java

```diff
@@ -16,28 +16,73 @@
 
 package com.google.cloud.tools.eclipse.appengine.libraries.model;
 
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
 import org.eclipse.core.runtime.IConfigurationElement;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 import org.mockito.Mockito;
 
 public class LibraryFactoryTest {
 
   private LibraryFactory factory = new LibraryFactory();
 
-  @Test
-  public void testCreate() throws LibraryFactoryException {
-    IConfigurationElement configuration = Mockito.mock(IConfigurationElement.class);
+  // todo is there a simple way to load this up from XML instead?
+  private IConfigurationElement configuration;
+  private IConfigurationElement[] libraryFiles = new IConfigurationElement[1];
+  private IConfigurationElement[] mavenCoordinates = new IConfigurationElement[1];
+
+  @Before
+  public void setUp() {
+    configuration = Mockito.mock(IConfigurationElement.class);
     Mockito.when(configuration.getAttribute("id")).thenReturn("guava");
+    
+    
+    libraryFiles[0] = Mockito.mock(IConfigurationElement.class);
+    Mockito.when(libraryFiles[0].getAttribute("mavenCoordinates")).thenReturn("mavenCoordinates");
+    Mockito.when(libraryFiles[0].getChildren()).thenReturn(new IConfigurationElement[0]);
+    mavenCoordinates[0] = Mockito.mock(IConfigurationElement.class);
+    Mockito.when(mavenCoordinates[0].getAttribute("groupId")).thenReturn("com.google.guava");
+    Mockito.when(mavenCoordinates[0].getAttribute("artifactId")).thenReturn("guava");
+    
     Mockito.when(configuration.getAttribute("group")).thenReturn("com.google.guava");
     Mockito.when(configuration.getName()).thenReturn("library");
     Mockito.when(configuration.getAttribute("siteUri"))
         .thenReturn(
         "https://cloud.google.com/storage/docs/reference/libraries#client-libraries-install-java");
-    Mockito.when(configuration.getChildren("libraryFile"))
-        .thenReturn(new IConfigurationElement[0]);
     Mockito.when(configuration.getChildren("libraryDependency"))
         .thenReturn(new IConfigurationElement[0]);
+    
+    Mockito.when(libraryFiles[0].getChildren("mavenCoordinates")).thenReturn(mavenCoordinates);
+    Mockito.when(libraryFiles[0].getName()).thenReturn("libraryFile");
+  }
+  
+  @Test
+  public void testCreate_useLatestVersion() throws LibraryFactoryException {
+    Mockito.when(configuration.getChildren("libraryFile")).thenReturn(libraryFiles);
+    
+    Library library = factory.create(configuration);
+    String version = library.getLibraryFiles().get(0).getMavenCoordinates().getVersion();
+    DefaultArtifactVersion artifactVersion = new DefaultArtifactVersion(version);
+    int majorVersion = artifactVersion.getMajorVersion();
+    Assert.assertTrue(majorVersion >= 22);
```

Comment:
latest version of Guava at the time of writing. (Now 23). When the configuration does not specify a specific version of the artifact, the maven repo is queried to find the latest available release version.